File: proxy/tests/script_engine/body_cascade.py
from roxy import Flow, Extension


class ChangeBody(Extension):
    def __init__(self, index: int):
        self.index = index

    def request(self, flow: Flow) -> None:
        t = flow.request.body.text or ""
        flow.request.body.text = f"{t} request{self.index}"

    def response(self, flow: Flow) -> None:
        t = flow.response.body.text or ""
        flow.response.body.text = f"{t} response{self.index}"


Extensions = [ChangeBody(1), ChangeBody(2)]

File: proxy/tests/script_engine/body_cascade.js
/// <reference path="../../script_libs/js/roxy.d.ts" />

/**
 * Make a BodyCascade extension that appends the given id.
 * @param {number} id
 * @returns {Extension}
 */
function makeBodyCascade(id) {
  return {
    request(flow) {
      flow.request.body.text = (flow.request.body.text ?? "") + " request" + id;
    },
    response(flow) {
      flow.response.body.text = (flow.response.body.text ?? "") + " response" + id;
    }
  };
}

globalThis.extensions = [
  makeBodyCascade(1),
  makeBodyCascade(2),
];
